DataFormats/TrackReco/interface/SiPixelTrackProbQXY.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SiPixelTrackProbQXY_H
+#define SiPixelTrackProbQXY_H
+
+#include <cstdint>
+#include <vector>
+
+#include "DataFormats/Common/interface/Association.h"
+#include "DataFormats/Common/interface/RefVector.h"
+
+namespace reco {
+  // Class defining the combined charge and shape probabilities for tracks that go through the SiPixel detector
+  class SiPixelTrackProbQXY {
+  public:
+    SiPixelTrackProbQXY() {}
+
+    SiPixelTrackProbQXY(float probQonTrack, float probXYonTrack, float probQonTrackNoL1, float probXYonTrackNoL1)
+        : probQonTrack_(probQonTrack),
+          probXYonTrack_(probXYonTrack),
+          probQonTrackNoL1_(probQonTrackNoL1),
+          probXYonTrackNoL1_(probXYonTrackNoL1) {}
+
+    // Return the combined charge probabilities for tracks that go through the SiPixel detector
+    float probQonTrack() const { return probQonTrack_; }
+
+    // Return the combined shape probabilities for tracks that go through the SiPixel detector
+    float probXYonTrack() const { return probXYonTrack_; }
+
+    // Return the combined charge probabilities for tracks that go through the SiPixel detector
+    // This version now excludes layer 1 which was known to be noisy for 2017/2018
+    float probQonTrackNoL1() const { return probQonTrackNoL1_; }
```

Comment:
```suggestion
    float probQonTrackNoBPixL1() const { return probQonTrackNoBPixL1_; }
```

?

Reply:
I'm not sure I understand the reasoning behind this? FPix "disk" are not called "layers", so is it so that we distinguish from the strips L1? The class is called `SiPixel....` so I' not sure any further specializations are needed.

Reply:
"L1" has not an unambiguous meaning in CMS. Would `probQonTrackNoLayer1` be a better option?

Reply:
Ok, I went with the `probQonTrackNoLayer1` option.